### PR TITLE
IAT-1184

### PR DESCRIPTION
**Pull request by hbgwhite**
Support for stacked Spanish translations for EQ, GI, MI, MC, MS, SA, TI, WER, STIM, and TUT item types.

### CODE REVIEW

**Comment by hbgwhite** [comment]
https://jira.fairwaytech.com/browse/IAT-1184

File: src/main/java/org/opentestsystem/ap/common/mapper/MiMapper.java [hunk 1]
```diff
@@ -96,13 +63,42 @@ public static String generateSaaifQtiSingleLine(final MiItem item, final MiItemC
...
+                                           final String language) {
         final Map<String, Object> input = new HashMap<>();
 
-        input.put(MODEL_KEY_COLUMNS, getColumnLabels(item, content, itemContext));
-        input.put(MODEL_KEY_ROWS, getRowLabels(item, content, itemContext));
+        final List<String> columnLabels = getColumnLabels(item, item.getCore().getEn(), itemContext);
+        final List<String> rowLabels = getRowLabels(item, item.getCore().getEn(), itemContext);
+        final List<String> columnLanguageLabels = new ArrayList<>();
+        final List<String> rowLanguageLabels = new ArrayList<>();
+        if (!language.equalsIgnoreCase("LANG_ENU")) {
```
Comment (alex-ponce) [inline src/main/java/org/opentestsystem/ap/common/mapper/MiMapper.java]: This condition can be changed for the LANG_ENU constant.

File: src/main/java/org/opentestsystem/ap/common/mapper/MiMapper.java [hunk 2]
```diff
@@ -129,20 +126,22 @@ public static String generateSaaifQrxSingleLine(final MiItem item, final MiItemC
...
+                                          final MiItemContent content,
+                                          final StringAssembler assembler) {
         final Map<String, Object> input = new HashMap<>();
 
         List<String> answers = getCorrectAnswerMatrix(item, content);
 
         input.put(ITEM_ID_KEY, item.getId());
         input.put(MI_CORRECT_RESPONSES_KEY, answers);
-        input.put(MI_POINTS_PER_ANSWER_KEY, (1.0/answers.size()));
+        input.put(MI_POINTS_PER_ANSWER_KEY, (1.0 / answers.size()));
```
Comment (alex-ponce) [inline src/main/java/org/opentestsystem/ap/common/mapper/MiMapper.java]: divide by zero vulnerability?

File: src/main/java/org/opentestsystem/ap/common/mapper/MiMapper.java (hunk 1 above)
Comment (hbgwhite) [reply src/main/java/org/opentestsystem/ap/common/mapper/MiMapper.java]: Good catch!

**Pull request merged**